specs/latest/2.0/index.html

```diff
@@ -1952,18 +2105,24 @@ <h2 class="no-toc">Table of contents</h2>
 
     <h4><a name="readpixels">Reading back pixels</a></h4>
 
-    <p>Pixels in the current framebuffer can be read back into an ArrayBufferView object or a WebGLBuffer bound to the <code>PIXEL_PACK_BUFFER</code> target.</p>
+    <p>Pixels in the current framebuffer can be read back into an BufferDataSource
+       object or a WebGLBuffer bound to the <code>PIXEL_PACK_BUFFER</code> target.
     <p>Only differences from <a href="../1.0/index.html#readpixels">Reading back pixels</a> in WebGL 1 are described here.</p>
     <dl class="methods">
-        <dt class="idl-code">void readPixels(GLint x, GLint y, GLsizei width, GLsizei height, GLenum format, GLenum type, ArrayBufferView? pixels)
-            <span class="gl-spec">(<a href="http://www.khronos.org/registry/gles/specs/3.0/es_spec_3.0.4.pdf#nameddest=section-4.3.2">OpenGL ES 3.0 &sect;4.3.2</a>, <a class="nonnormative" href="http://www.khronos.org/opengles/sdk/docs/man3/html/glReadPixels.xhtml">man page</a>)</span>
+        <dt class="idl-code">
+          void readPixels(GLint x, GLint y, GLsizei width, GLsizei height,
+                          GLenum format, GLenum type, BufferDataSource dstData,
+                          optional unsigned long dstOffset = 0)
+          <span class="gl-spec">(<a href="http://www.khronos.org/registry/gles/specs/3.0/es_spec_3.0.4.pdf#nameddest=section-4.3.2">OpenGL ES 3.0 &sect;4.3.2</a>, <a class="nonnormative" href="http://www.khronos.org/opengles/sdk/docs/man3/html/glReadPixels.xhtml">man page</a>)</span>
         </dt>
         <dd>
           <p>If a WebGLBuffer is bound to the <code>PIXEL_PACK_BUFFER</code> target, generates an <code>INVALID_OPERATION</code> error.</p>
           <p>
               If <a href="#PIXEL_STORE_PARAM_CONSTRAINTS">pixel store parameter constraints</a> are not met,
               generates an <code>INVALID_OPERATION</code> error.
           </p>
+          <p>If <code>dstData</code> doesn't have enough data starting at
+             <code>dstOffset</code>, generate <code>INVALID_OPERATION</code>.
```

Comment:
Perhaps: if there isn't enough space in dstData starting at dstOffset for the result of the read operation, generates an INVALID_OPERATION error.